### PR TITLE
Add more CUB transform benchmarks 

### PR DESCRIPTION
**Pull request by bernhardmgruber**
This PR adds a bunch of new benchmarks for CUB transform. I still have to figure out whether all benchmarks are necessary, because I don't want to cover redundant aspects.

~~Turns out, we cannot run this using tuning:~~
```
#### ERROR exception occured while running cub.bench.transform.other: 'All values in the dictionary are not equal. First value: {'OffsetT{ct}': ['I32', 'I64']} All values: [{'OffsetT{ct}': ['I32', 'I64']}, {'OffsetT{ct}': ['I32', 'I64']}, {'OffsetT{ct}': ['I32', 'I64']}, {'Heaviness{ct}': ['32', '64', '128', '256']}]'
```
@gevtushenko do all sub benchmarks within the same benchmark file must have the same axes?

Edit: I split the benchmarks into separate files now and this resolved the issue.

### CODE REVIEW

File: cub/benchmarks/bench/transform/other.cu [hunk 1]
```diff
@@ -0,0 +1,194 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+// %RANGE% TUNE_THREADS tpb 128:1024:128
+// %RANGE% TUNE_ALGORITHM alg 0:1:1
+
+// keep checks at the top so compilation of discarded variants fails really fast
+#if !TUNE_BASE
```
Comment (miscco) [inline cub/benchmarks/bench/transform/other.cu:8]: Nitpick: Can we find a better name than `other`

**Review by miscco — APPROVED** [review]

File: cub/benchmarks/bench/transform/other.cu (hunk 1 above)
Comment (bernhardmgruber) [reply cub/benchmarks/bench/transform/other.cu:8]: I honestly still need to grasp whether I should just split each benchmark into a dedicated file. Then those would have better names.

Comment (bernhardmgruber) [reply cub/benchmarks/bench/transform/other.cu]: I split the benchmarks and gave them better names.

**Comment by bernhardmgruber** [comment]
Non-serious run on my workstation:
image
Looks like `compare_complex` and `heavy` are compute bound (as expected) and thus don't reach SOL memory bandwidth. I expected fibonacci (thread divergence) and vertex_transform (more compute than babelstream) to be slower. Let's see what happens on a different GPU.

**Comment by bernhardmgruber** [comment]
On H100:
image
Here, the impact of thread divergence (fib), compute intensity (compare_complex) and compute + register pressure (heavy) are more starkly pronounced. At this point I feel the `vertex_transform` example is not giving us any new insight.

**Review by gevtushenko — APPROVED** [review]

**Pull request merged**